Gems/ROS2/Code/Source/VehicleDynamics/DriveModels/AckermannDriveModel.cpp

```diff
@@ -164,4 +140,9 @@ namespace ROS2::VehicleDynamics
         m_disabled = isDisabled;
     }
 
+    VehicleModelLimits* AckermannDriveModel::GetVehicleLimits(){
```

Comment:
It has to be done this way - the m_limit has to be member of derivative class because of reflection. We cannot reflect smart pointers or raw pointers.

Reply:
Hower, I've made everything protected. It still can dangle but in the scope of the base class, what is acceptable.